x/msgfee/init.go

```diff
@@ -0,0 +1,43 @@
+package msgfee
+
+import (
+	"fmt"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+)
+
+// Initializer fulfils the Initializer interface to load data from the genesis
+// file
+type Initializer struct{}
+
+var _ weave.Initializer = (*Initializer)(nil)
+
+// FromGenesis will parse initial account info from genesis and save it to the
+// database
+func (*Initializer) FromGenesis(opts weave.Options, db weave.KVStore) error {
+	type msgfee struct {
+		MsgPath string    `json:"msg_path"`
+		Fee     coin.Coin `json:"fee"`
+	}
+	var fees []*msgfee
```

Comment:
curiosity question:
why do we need `[]*msgFee` rather than `[]msgFee`?

Reply:
That is a great question!

This is because when iterating over a slice of non-pointers "current" value is always overwritten in each step. This means that if you reference it you will end up with the same value everywhere.

https://play.golang.org/p/diWYP3ElXmF

Reply:
Nice tip.

It took me a bit to understand where the big came from even after seeing it run.